### PR TITLE
Add the guest_search, guest_uuid, host_uuid, host_hwuuid functions

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
Add the guest_search, guest_uuid, host_uuid, host_hwuuid functions

- [x] Add log info for ssh command
- [x] Add `guest_search`, `guest_uuid`, `host_uuid`, `host_hwuuid` functions
- [x] Add/update docstring
- [x] Update the `ConvertToJSon` method
- [x] Use the` f-string` instead of `.format`

**Test Result**
[IN]
```
    server = '10.73.5.xxx'
    admin_user = 'admxxxx'
    admin_passwd = 'Welxxxxx'
    ssh_user = 'Admxxxxxr'
    ssh_passwd = 'Wexxxx'
    guest_name = '8.3_Server_x86_64'
    cli = PowerCLI(server, admin_user, admin_passwd, ssh_user, ssh_passwd)
    cli.info()
    msgs = cli.guest_search(guest_name)
```
[OUT]
```
msgs =
{'guest_name': '8.3_Server_x86_64', 'guest_ip': '10.xx.x.xxx', 'guest_uuid': '423f8145-339d-bcf0-7cdd-e94ba4a687dd', 'guest_state': 2, 'guest_cpu': 1, 'esx_ip': 'xx.xx.x.xxx', 'esx_uuid': '67619929-b96b-4724-983b-b299d26e7e31', 'esx_hwuuid': 'host-44', 'esx_hostname': ''xx.xx.x.xxx', 'esx_version': '6.7.0', 'esx_state': 0, 'esx_cpu': 2, 'esx_cluster': 'hkx-test'}
```

### CODE REVIEW

**Review by eko999 — COMMENTED** [review]
That's great, I have a suggestion from me is could we have an uniform string format, how about the f-strings?

**Comment by hkx303** [comment]
Please DO NOT MERGE for now as I still meet a blocked issue.

**Comment by hkx303** [comment]
> That's great, I have a suggestion from me is could we have an uniform string format, how about the f-strings?

Yeah, I also have the idea to do it as f-string has a fast and readable advantage. :)

**Comment by hkx303** [comment]
Remove the `do not merge` label as I have fixed the issue.

File: hypervisor/virt/esx/powercli.py
Comment (eko999) [inline hypervisor/virt/esx/powercli.py:73]: As said, we may define a uniform guest state which matched with the mapping info in rhsm.log.

**Review by Junefen — APPROVED** [review]
Now it seem that the output is able to meet the demand for virtwho-test.

But it's better for guest_search() to provide searching by guest_name or guest_ip, I guess there will be requirement from some other users. But maybe we can perfect it in the future.

**Comment by hkx303** [comment]
> Now it seem that the output is able to meet the demand for virtwho-test.
> 
> But it's better for guest_search() to provide searching by guest_name or guest_ip, I guess there will be requirement from some other users. But maybe we can perfect it in the future.

Thanks for the suggestion, that is a good point, will consider it in the future :)

File: hypervisor/virt/esx/powercli.py
Comment (hkx303) [reply hypervisor/virt/esx/powercli.py:73]: Will try some cases to see if the guest state meets our requirements

Comment (hkx303) [reply hypervisor/virt/esx/powercli.py:73]: Hey Eko, I prefer to do it with another PR as I still need some other functions to add for now and I still need some time...
what is your thought? Merge this PR if you are OK so that we can move on :)

Comment (eko999) [reply hypervisor/virt/esx/powercli.py:73]: Cool, I will merge it.

**Review by eko999 — APPROVED** [review]

**Pull request merged**
